### DISCUSSION TITLE
[Feature] How to view zap Caller in skywalking-ui

### DISCUSSION THREAD

**Discussion by Capricornwqh** [apache/skywalking#12747] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

...
"caller":"zap/options.go:212",
"func":"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap.DefaultMessageProducer",
"msg":"finished unary call with code OK"
...

"SW_AGENT_LOG_REPORTER_LABEL_KEYS": "caller,func" ,
SW_AGENT_LOG_REPORTER_LABEL_KEYS is not work
![1](https://github.com/user-attachments/assets/0bdfcc46-6885-411f-9ec9-76f0d0438512)

### Are you willing to submit a pull request to implement this on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by mrproliu** [comment]
We only support reporting the labels by self-customization. If you want to add the caller, you must change the [reporter code in there](https://github.com/apache/skywalking-go/blob/main/tools/go-agent/instrument/logger/frameworks/zap_core.go#L29-L44), the caller is inside the entry parameter.

**Reply by Capricornwqh** [reply]
I hope we can support the labels in the future, as both the caller and the function are crucial for troubleshooting, especially during the testing phase.

**Reply by wu-sheng** [reply]
@Capricornwqh Open source project is pending on volunteers' contributions, which includes yours. If you want to propose something, try to implement them and submit through a pull request.

**Reply by Capricornwqh** [reply]
```
func ReportLogFromZapEntry(entry *zapcore.CheckedEntry, fields, needs []zapcore.Field, tracingContext interface{},
	tracingContextField *zapcore.Field, reporterEnable, logEnable bool, reportLabelsKeys []string) []zapcore.Field {
	if reporterEnable && tracingContext != nil {
		labels := make(map[string]string, len(reportLabelsKeys))
		for _, key := range reportLabelsKeys {
			for _, f := range fields {
				if f.Key == key {
					if k, v := generateLabelKeyValueFromField(f); k != "" {
						labels[k] = v
					}
				}
			}
		}
		for _, f := range needs {
			if k, v := generateLabelKeyValueFromField(f); k != "" {
				labels[k] = v
			}
		}
		if entry.Caller.Defined {
			labels["FILE"] = fmt.Sprintf("%s:%d", entry.Caller.File, entry.Caller.Line)
			labels["FUNC"] = entry.Caller.Function
		}

		ReportLog(tracingContext, entry.Time, entry.Level.String(), entry.Message, labels)
	}
	if logEnable && tracingContextField != nil {
		fields = append(fields, *tracingContextField)
	}
	return fields
}
```

i change this code, it has worked
![image](https://github.com/user-attachments/assets/0e90fe98-e6e4-446f-83ff-88e540a0d403)

**Reply by wu-sheng** [reply]
If you want to propose this change, please submit it through a pull request. Then the maintainers could review your codes.

**Reply by Capricornwqh** [reply]
the main branch has an error, but 0.5.0 is ok
```
# github.com/apache/skywalking-go/toolkit/trace
$WORK/b1153/skywalking_enhance_trace_add_event_intercepter.go:26:31: undefined: EventType
$WORK/b1153/skywalking_enhance_trace_async_add_event_intercepter.go:33:30: undefined: EventType
```
